src/include/api.h

```diff
@@ -10,48 +10,66 @@
 /*
  * Capture cases where a single session handle is used by multiple threads in parallel. The check
  * isn't trivial because some API calls re-enter via public API entry points and the session with ID
- * 0 is the default session in the connection handle which can be used across multiple threads. It
- * is safe to use the reference count without atomic operations because the reference count is only
- * tracking a thread re-entering the API.
+ * 0 is the default session in the connection handle which can be used across multiple threads.
  */
-#define WT_SINGLE_THREAD_CHECK_START(s)                                                   \
-    {                                                                                     \
-        uintmax_t __tmp_api_tid;                                                          \
-        __wt_thread_id(&__tmp_api_tid);                                                   \
-        WT_ASSERT(s, (s)->id == 0 || (s)->api_tid == 0 || (s)->api_tid == __tmp_api_tid); \
-        if ((s)->api_tid == 0) {                                                          \
-            (s)->api_tid = __tmp_api_tid;                                                 \
-            WT_FULL_BARRIER();                                                            \
-        }                                                                                 \
-        ++(s)->api_enter_refcnt;                                                          \
+#define WT_SINGLE_THREAD_CHECK_START(s)                                           \
+    {                                                                             \
+        uintmax_t __tmp_api_tid;                                                  \
+        __wt_thread_id(&__tmp_api_tid);                                           \
+                                                                                  \
+        /*                                                                        \
+         * Only a single thread should use this session at a time. It's ok        \
+         * (but unexpected) if different threads use the session consecutively,   \
+         * but concurrent access is not allowed. Verify this by having the thread \
```

Comment:
You dropped the work "take" here.

Reply:
Fixed in [ef0cc25](https://github.com/wiredtiger/wiredtiger/pull/9810/commits/ef0cc256037ab680eea541474d13be6624b1378c)